xenmgr/Vm/Actions.hs

```diff
@@ -878,7 +878,8 @@ bootVm config reboot
                 then do 
                   if reboot 
                     then do liftIO $ Xl.signal uuid
-                    else do liftIO $ Xl.start uuid --we start paused by default
+                    else do tapenv <- tapEnvForVm uuid
```

Comment:
Encrypted VHDs need TAPDISK3_CRYPTO_KEYDIR set.  Initially I wrote block-tap script to use xec-vm to populate it.  This made the call tree look kinda like:
```
xenmgr
 `- xl
    `- block-tap
        `- TAPDISK3_CRYPTO_KEYDIR=$( xec-vm get crypto-key-dirs )
        `- tap-ctl
            `- tapdisk - Needs TAPDISK3_CRYPTO_KEYDIR
```

Since xec-vm is calling back to xenmgr, I thought we could just have xenmgr populate TAPDISK3_CRYPTO_KEYDIR and bypass that extra dbus call.

I think block-tap populating TAPDISK3_CRYPTO_KEYDIR is clearer and more straightforward.  I worry about the subtlety of relying on TAPDISK3_CRYPTO_KEYDIR being populated transitively through xl.  But it seemed roundabout for block-tap to call back to xenmgr which invoked xl in the first place, so I eliminated that.  I had to do something - ¯\\_(ツ)_/¯

Maybe block-tap should check for an empty/unset TAPDISK3_CRYPTO_KEYDIR and only make the xec-vm call in that case?